pkg/search/search.go

```diff
@@ -0,0 +1,137 @@
+package search
+
+import (
+	"fmt"
+)
+
+type Searcher interface {
+	Definition(word string) (string, error)
+	Definitions(words []string) (string, error)
```

Comment:
A good rule of thumb - If you have an interface called `Searcher`, try to give it a single method called `search`. Note that you can make `search` take `words []string` which covers the single-word case as well. This will let you have a one-method interface, which is ideal. Also, consider the return value here - is a string sufficient here, or would you have a use for a struct that could potentially let you give the answer some more structure?

Reply:
That makes sense! 
For the return value, I'm still not sure. I imagine it this way atm: the gui package just wants to output the definition. It just prints a string to the graphical interface. 
Therefore, should the gui package receive a more detailed response? Which would entail learning more about an object? Or should it receive a string that it just prints that out to the user?

Reply:
Think of it in another way - the GUI _knows_ it's using a "Search" capability. But the "Search" capability has no idea who it's being used by. Therefore, it wouldn't make sense for the "Search" capability to structure its output based on its consumer. Instead, it'd make more sense for the GUI to take the output of the search and massage it however it pleases.

Reply:
Oh. Well that completely makes sense to me now :+1: . Will make the necessary changes